plugins/couchbase/couchbaseembed.cpp

```diff
@@ -1002,7 +1035,17 @@ namespace couchbaseembed
         }
 
         StringBuffer fullXPath;
-        constructNewXPath(fullXPath, xpath);
+
+        if (strncmp(xpath, "<set element>", 13) == 0 && !m_pathStack.empty() && m_pathStack.back().nodeType == CPNTSet)
```

Comment:
minor, can we test "!m_pathStack.empty()", or "m_pathStack.back().nodeType == CPNTSet" before we perform the strncmp? In some cases it could will prevent the expensive string test when unnecessary.